### ISSUE TITLE
Commands: cannot identify all context from service container

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/EntityFramework#5156]
At design-time, we relied on the presence of `DbContextOptions` to figure our which types of context have been added in startup via `AddDbContext<T>()`. With recent changes, this means only one type of DbContextOptions is ever present.

## Repro
```
serviceCollection
    .AddEntityFrameworkSqlite()
    .AddDbContext<Context1>(getOptions1)
    .AddDbContext<Context2>(getOptions2);
```
Then make a call to DbContextOperations.GetContextTypes();
## Expected
DbContextOperations.GetContextTypes() should return Context1 and Context2

## Actual
DbContextOperations.GetContextTypes() only returns Context1

**Comment by natemcmaster** [comment]
cc @ajcvickers

**Comment by divega** [comment]
@ajcvickers I told @natemcmaster to unblock himself by replacing the call to `TryAddSingleton<DbContextOptions>()` inside `AddDbContext<TContext>()` with `AddSingleton<DbContextOptions>()` so that all `DbContextOptions<TContext>` get registered as `DbContextOptions`. 

AFAICS the only observable changes from what we have now should be:

1. `GetService<DbContextOptions>()` will return the last one registered instead of the first one.
2. `GetServices<DbContextOptions>()` should return them all (which is what migrations discovery logic needs)

Can you think of any other drawback?

**Comment by natemcmaster** [comment]
One drawback I've found is https://github.com/aspnet/EntityFramework/issues/4994. We should put https://github.com/aspnet/EntityFramework/pull/5017 in RC2

**Comment by ajcvickers** [comment]
@divega @natemcmaster For people using AddDbContext in a normal way there should be not drawbacks. #4994 still exists in either case, but as Diego said the context type that gets the wrong connection string will be switched.

The only drawback I am aware of (and the reason it was a TryAdd) is that if somebody else is registering DbContextOptions outside of AddDbContext and then calls AddDbContext then we will clobber what they registered. We make an effort not to do that in our AddXxx methods, but if we have to then I don't think it is a big deal.

**Comment by divega** [comment]
Thanks @ajcvickers. I agree it is not a big deal. 

I think @natemcmaster hit #4994 almost immediately and that is what he means. I agree it is a shame that the improved exception are not in RC2. We have seen a few customers hitting it as well. Good XML docs would help too.

**Comment by natemcmaster** [comment]
Fixed in #5183
